File: winter intersem1/gdbLearning/josephus.cpp
// { Driver Code Starts
#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends
/*You are required to complete this method */

class Solution
{
    int x=0;
public:
    int josephus(int n, int k)
    {
        // Your code here
        if (k <= 0)
        {
            return n;
        }
        // int x;
        x++;
        josephus(--n, --k);
        return x+1;
    }
};

// { Driver Code Starts.

int main()
{

    int n, k;
    cin >> n >> k; // taking input n and k

    // calling josephus() function
    Solution ob;
    cout << ob.josephus(n, k) << endl;

    return 0;
} // } Driver Code Ends

File: summerSem/josephus.java
public class josephus {
    static int solution(int p, int s) {
        if (p == 1)
            return 1;
        return (solution(p - 1, s) + s - 1) % p + 1;
    }

    public static void main(String[] args) {
        int p = 9, s = 5;
        int rel = solution(p, s);
        System.out.println("safe position : " + rel);
    }
}